analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -241,20 +243,72 @@ class Pub(
             log.info { "Successfully read lockfile." }
 
             val parsePackagesResult = parseInstalledPackages(lockFile, labels)
+            val dependenciesScope = parseScope("dependencies", manifest, lockFile, parsePackagesResult.packages, labels)
+
+            if (containsFlutterSdk(workingDir)) {
+                // Run Gradle dependency resolution. This is required because for Flutter projects, Pub needs to run
+                // first to generate the local.properties file by using `flutter pub get`.
+                val gradleFactory = Gradle.Factory()
+                val gradleDefinitionFiles =
+                    findManagedFiles(analysisRoot, listOf(gradleFactory)).getOrDefault(gradleFactory, emptyList())
+
+                if (gradleDefinitionFiles.isNotEmpty()) {
+                    println("Found ${gradleDefinitionFiles.size} ${gradleFactory.managerName} project(s) at:")
+
+                    gradleDefinitionFiles.forEach {
+                        val relativePath = it.toRelativeString(workingDir).takeIf { it.isNotEmpty() } ?: "."
+
+                        println("\t$relativePath")
+                    }
+                }
+
+                log.info { "Running Gradle analysis for Flutter project." }
+
+                // TODO: This ignores the AnalyzerCommand's `--package-managers` option and always runs Gradle for
+                //  Flutter projects.
```

Comment:
Indentation is off. But anyway, I believe we drop this TODO, as I'd argue that anyone who enabled Pub but disabled Gradle (or enabled Pub only) still wants Gradle projects as part of Pub projects to be analyzed.

Reply:
Removed. But for future reference: How should this be indented?
If both lines are are indented on the same level I don't get IDE highlighting for the whole TODO.

Reply:
See existing `TODO` statements 😉 All lines have to start their text in the column of the "T" from "This".